eoscommunity/Eden
# Add Signing session transactions example page

A page with examples of how a session key can be created and how it can be used to sign a transaction was created in this PR. For that, it was needed to create the frontend functionality to create and store the session key (using IndexedDb) and the glue between the frontend and eden box.

To test it:
1. Spin up an ephemeral chain
1. Go to https://localhost:3000/sessions
1. Open the inspector to observe the console
1. Click on create session key
1. Click on induct Ahab to see that we are using the created sessionkey
1. You can see the created induction on the membership page

Next steps:
- Work on getting the proper sequence number dynamically (persist it on the microchain and get it from the useCurrentMember query)
- Work on the UI flows to create session keys, detected when it's created and use it on allowed transactions in the UI
- Work on the re-generate session key flow when it expires (it has a limit of 90 days in the contract and in the current creation we are actually setting it to 30 days)
- Migrate from eosjs to SubtleCrypto

In the meanwhile:
- We could go ahead and create an eden payer account in Wax testnet and EOS mainnet and set the proper env configs for the eden boxes in these environments.
